### PR TITLE
chore: update go-octopusdeploy library reference

### PR DESCRIPTION
**Pull request by benPearce1**
create project is still working
image

project view is still working
image

### CODE REVIEW

File: pkg/cmd/release/create/create.go [hunk 1]
```diff
@@ -820,7 +820,7 @@ func AskQuestions(octopus *octopusApiClient.Client, stdout io.Writer, asker ques
 	// we always need the deployment process, so we can prompt for package version overrides (or know that there aren't any packages, so it doesn't matter)
 
 	var gitReferenceKey string
-	if selectedProject.PersistenceSettings.GetType() == "VersionControlled" {
+	if selectedProject.PersistenceSettings.Type() == "VersionControlled" {
```
Comment (slewis74) [inline pkg/cmd/release/create/create.go:823]: there is a public const defined in the Go client for this, rather than comparing to a magic string

**Review by slewis74 — APPROVED** [review]
Looks good. Shall we add support for Protected Branch Patterns in a separate PR? Probably easier.

File: pkg/cmd/project/create/create_opts.go [hunk 2]
```diff
@@ -126,11 +126,11 @@ func (co *CreateOptions) buildVersionControlSettings() (*projects.VersionControl
 		return nil, err
 	}
 
-	vcs := projects.NewVersionControlSettings(getBasePath(co), credentials, getGitBranch(co), GitPersistenceType, url)
+	vcs := projects.NewGitPersistenceSettings(getBasePath(co), credentials, getGitBranch(co), []string{}, url)
```
[the remark this answers is not in the document]
Comment (benPearce1) [reply pkg/cmd/project/create/create_opts.go:129]: That sucks

File: pkg/cmd/release/create/create.go (hunk 1 above)
Comment (benPearce1) [reply pkg/cmd/release/create/create.go:823]: yeah, I removed an unused one, will update to use the shared one

Comment (benPearce1) [reply pkg/cmd/release/create/create.go:823]: fixed

**Pull request merged**